pandas/io/parsers/base_parser.py

```diff
@@ -503,7 +505,11 @@ def _agg_index(self, index, try_parse_dates: bool = True) -> Index:
             if isinstance(clean_dtypes, dict) and self.index_names is not None:
                 cast_type = clean_dtypes.get(self.index_names[i], None)
 
-            try_num_bool = not (cast_type and is_string_dtype(cast_type))
+            conv = False
+            if isinstance(converters, dict) and self.index_names is not None:
```

Comment:
Could combine with the above check as  `if self.index_names is not None:` first and then the 2 `isinstance(arg, dict)` checks

Reply:
Thx, this looks better, also renamed the variable